sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -1368,22 +1441,62 @@ def plot(
         self.deciles_vlines_ = np.empty_like(self.axes_, dtype=object)
         self.deciles_hlines_ = np.empty_like(self.axes_, dtype=object)
 
-        for pd_plot_idx, (axi, feature_idx, pd_result) in enumerate(
-            zip(self.axes_.ravel(), self.features, self.pd_results)
+        for pd_plot_idx, (axi, feature_idx, pd_result, kind_plot) in enumerate(
+            zip(self.axes_.ravel(), self.features, self.pd_results, kind)
         ):
             avg_preds = None
             preds = None
             feature_values = pd_result["values"]
-            if self.kind == "individual":
+            if kind_plot == "individual":
                 preds = pd_result.individual
-            elif self.kind == "average":
+            elif kind_plot == "average":
                 avg_preds = pd_result.average
-            else:  # kind='both'
+            else:  # kind_plot == 'both'
                 avg_preds = pd_result.average
                 preds = pd_result.individual
 
             if len(feature_values) == 1:
+                # define the line-style for the current plot
+                default_line_kws = {
+                    "color": "C0",
+                    "label": "average" if kind_plot == "both" else None,
+                }
+                if kind_plot in "individual":
+                    default_ice_lines_kws = {"alpha": 0.3, "linewidth": 0.5}
+                    default_pd_lines_kws = {}
+                elif kind_plot in "both":
+                    # by default, we need to distinguish the average line from
+                    # the individual lines via color and line style
+                    default_ice_lines_kws = {
+                        "alpha": 0.3,
+                        "linewidth": 0.5,
+                        "color": "tab:blue",
+                    }
+                    default_pd_lines_kws = {
+                        "color": "tab:orange",
+                        "linestyle": "--",
+                    }
```

Comment:
Instead of deleting `ice_lines_kw["label"] here, I think we can set `label` here:

```suggestion
                    default_pd_lines_kws = {
                        "color": "tab:orange",
                        "linestyle": "--",
                        "label": "average"
                    }
```

Reply:
it is not enough because `"label"` is part of `default_line_kws`

Reply:
The problem comes if a user provides a `"label"` key when calling `plot` to overwrite the label. Then we need to delete the key in some cases.

Reply:
I forgot to finish the thought. I thought we could leave the label out of `default_line_kws` too, since it only applies to `default_pd_lines_kws`.

> The problem comes if a user provides a "label" key when calling plot to overwrite the label. Then we need to delete the key in some cases.

I think I see what you mean. If `line_kw` has a "label", then we want to remove the label for ice plots?

Scrolling up I see this is the same logic we had before, so I am okay with keeping it as is.

Reply:
Yep exactly